presto-kafka/src/test/java/io/prestosql/plugin/kafka/TestKafkaIntegrationSmokeTest.java

```diff
@@ -160,6 +162,119 @@ private KafkaTopicFieldDescription createOneFieldDescription(String name, Type t
         return new KafkaTopicFieldDescription(name, type, mapping, null, dataFormat, null, false);
     }
 
+    @Test(dataProvider = "testJsonDateTimeFormatsDataProvider")
+    public void testJsonDateTimeFormats(JsonDateTimeTestCase testCase)
```

Comment:
how is logic here different from `testRoundTripAllFormats`?
Can we make `testRoundTripAllFormats` and `testJsonDateTimeFormats` share testcase definition.

Reply:
yes, that's the plan when the raw encoder pr merges #4417